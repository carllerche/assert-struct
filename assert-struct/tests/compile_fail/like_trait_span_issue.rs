use assert_struct::assert_struct;

#[derive(Debug)]
struct Expr;

#[derive(Debug)]
struct QuerySql {
    filter: Expr,
}

fn user_id_string() -> String {
    "123".to_string()
}

fn main() {
    let query_sql = QuerySql {
        filter: Expr,
    };

    // This should fail with a type error pointing to the specific Like pattern expression
    // The error should point to `user_id_string()` not the entire macro call
    assert_struct!(query_sql, QuerySql {
        filter: =~ user_id_string(),  // Type error: Expr doesn't implement Like<String>
        ..
    });
}