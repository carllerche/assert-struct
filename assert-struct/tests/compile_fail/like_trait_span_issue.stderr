error[E0599]: no method named `like` found for reference `&Expr` in the current scope
  --> tests/compile_fail/like_trait_span_issue.rs:23:20
   |
23 |         filter: =~ user_id_string(),  // Type error: Expr doesn't implement Like<String>
   |                    ^^^^^^^^^^^^^^ method not found in `&Expr`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `like`, perhaps you need to implement it:
           candidate #1: `assert_struct::Like`
